contract/contracts/PixelDevsUkraineDonation.sol

```diff
@@ -58,32 +63,37 @@ contract PixelDevsUkraineDonation is ERC721Enumerable, ReentrancyGuard, Ownable
         minimumMintPrice = _newPrice;
     }
 
-    function mint(
-        uint256 tokenId
-    )
+    function mint()
         public
         payable
         nonReentrant
+        returns (uint256) 
     {
         require(minimumMintPrice <= msg.value, "Not enough MATIC sent");
-        _safeMint(msg.sender, tokenId);
+
+        _tokenIds.increment();
+        uint256 newTokenId = _tokenIds.current();
+
+        _safeMint(msg.sender, newTokenId);
 
         if (msg.value <= 25 ether) {
             donationType = DonationType.BRONZE;
-            emit LogTokenMinted(msg.sender, tokenId, "bronze");
+            emit LogTokenMinted(msg.sender, newTokenId, "bronze");
         } else if (msg.value <= 50 ether) {
             donationType = DonationType.SILVER;
-            emit LogTokenMinted(msg.sender, tokenId, "silver");
+            emit LogTokenMinted(msg.sender, newTokenId, "silver");
         } else if (msg.value <= 100 ether) {
             donationType = DonationType.GOLD;
-            emit LogTokenMinted(msg.sender, tokenId, "gold");
+            emit LogTokenMinted(msg.sender, newTokenId, "gold");
         } else if (msg.value <= 500 ether) {
             donationType = DonationType.DIAMOND;
-            emit LogTokenMinted(msg.sender, tokenId, "diamond");
+            emit LogTokenMinted(msg.sender, newTokenId, "diamond");
         } else {
             donationType = DonationType.PLATINUM;
-            emit LogTokenMinted(msg.sender, tokenId, "platinum");
+            emit LogTokenMinted(msg.sender, newTokenId, "platinum");
         }
+
+        return newTokenId;
```

Comment:
It turns out that we actually don't need to return the tokenId as this cannot be retrieved by the frontend after all. Apparently it's related to X block confirmations needed from my understanding. 

However it *can* read the events emitted by the function so I can simply use the LogTokenMinted event to retrieve the tokenId. So it's all good :-)